src/types.rs

```diff
@@ -51,7 +52,7 @@ pub enum EventCategory {
     CommandAndControl,
     Exfiltration,
     Impact,
-    HttpThreat,
+    HttpThreat, // Obsoleted. Do not use this category
```

Comment:
Instead of comment, [deprecated attribute](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute) could be more helpful.
```suggestion
    #[deprecated(note = "Users should use `EventCategory::Reconnaissance` instead")]
    HttpThreat, 
```

Reply:
After addressing the clippy warnings, it looks good to me.

Reply:
With the directive `#[deprecated(note = "...")]`, `HttpThreat` enum member always makes clippy error like following:
```rust
error: use of deprecated unit variant `types::EventCategory::HttpThreat`: Users should use `EventCategory::Reconnaissance` instead
  --> src/types.rs:56:5
   |
56 |     HttpThreat, // Obsoleted. Do not use this category
   |     ^^^^^^^^^^
   |
   = note: `-D deprecated` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(deprecated)]`
```

The directive to indicate that it is `deprecated` creates a deprecated clippy error. I haven't found a solution yet.

I changed it to comment now.

Reply:
Sorry, I didn't realize that proc macro would still cause the warning even with `#[allow(deprecated)]`. The following should do the trick.
```diff
diff --git a/src/types.rs b/src/types.rs
index 0a7cb57..0b41480 100644
--- a/src/types.rs
diff --git a/src/types.rs b/src/types.rs
index 0a7cb57..0b41480 100644
--- a/src/types.rs
+++ b/src/types.rs
@@ -1,3 +1,4 @@
+#![allow(deprecated)]
 use std::{cmp::Ordering, net::IpAddr, ops::RangeInclusive};
 
 use anyhow::Result;
@@ -52,7 +53,8 @@ pub enum EventCategory {
     CommandAndControl,
     Exfiltration,
     Impact,
-    HttpThreat, // Obsoleted. Do not use this category
+    #[deprecated(note = "Users should use `EventCategory::Reconnaissance` instead")]
+    HttpThreat,
 }
 
 #[derive(Deserialize)]
```